### DISCUSSION TITLE
Cannot find tsconfig.base.json

### DISCUSSION THREAD

**Discussion by iDVB** [nrwl/nx#13379] [General]
Can't seem to get off the ground with nx generators.
https://nrwlcommunity.slack.com/archives/CMFKWPU6Q/p1669299297492689

We already have a package based mono repo using nx. We have libs in `/packages/{lib_here}` and I want to use generator to create a new lib… I’ve never done this before….
I go to the root of the mono repo and type…
`nx generate @nrwl/js:library react-ui --project=packages --directory=./packages/react-ui`
and I get ….
`Cannot find tsconfig.base.json`
I then cd into packages and try again…same msg.

**Comment by iDVB** [comment]
Ended up being that we're migrating from a packaged based repo to an integrated one and there was no `tsconfig.base.json` so I just created one...

```
{
  "compileOnSave": false,
  "compilerOptions": {
    "rootDir": ".",
    "sourceMap": true,
    "declaration": false,
    "moduleResolution": "node",
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "importHelpers": true,
    "target": "es2015",
    "module": "esnext",
    "lib": ["es2017", "dom"],
    "skipLibCheck": true,
    "skipDefaultLibCheck": true,
    "baseUrl": ".",
    "paths": {}
  },
  "exclude": ["node_modules", "tmp"]
}

```

**Reply by squidjam** [reply]
VSCode complained (actually it was the linting) requiring me to add :
```
...
    "strict": true,
    "forceConsistentCasingInFileNames": true,
...
```
to the `compilerOptions` entry.
